Client/WriteApiAsync.cs

```diff
@@ -0,0 +1,304 @@
+        /// <param name="precision">specifies the precision for the unix timestamps within the body line-protocol</param>
+        /// <param name="measurements">specifies Measurements to write into bucket</param>
+        /// <typeparam name="TM">measurement type</typeparam>
+        public async Task WriteMeasurementsAsync<TM>(string bucket, string org, WritePrecision precision,
+                        List<TM> measurements)
+        {
+            Arguments.CheckNonEmptyString(bucket, nameof(bucket));
+            Arguments.CheckNonEmptyString(org, nameof(org));
+            Arguments.CheckNotNull(precision, nameof(precision));
+
+            foreach (var measurement in measurements) await WriteMeasurementAsync(bucket, org, precision, measurement);
+        }
+
+        /// <summary>
+        /// Write Measurements into specified bucket.
+        /// </summary>
+        /// <param name="precision">specifies the precision for the unix timestamps within the body line-protocol</param>
+        /// <param name="measurements">specifies Measurements to write into bucket</param>
+        /// <typeparam name="TM">measurement type</typeparam>
+        public async Task WriteMeasurementsAsync<TM>(WritePrecision precision, params TM[] measurements)
+        {
+            Arguments.CheckNotNull(precision, nameof(precision));
+
+            await WriteMeasurementsAsync(_options.Bucket, _options.Org, precision, measurements);
+        }
+
+        /// <summary>
+        /// Write Measurements into specified bucket.
+        /// </summary>
+        /// <param name="bucket">specifies the destination bucket for writes</param>
+        /// <param name="org">specifies the destination organization for writes</param>
+        /// <param name="precision">specifies the precision for the unix timestamps within the body line-protocol</param>
+        /// <param name="measurements">specifies Measurements to write into bucket</param>
+        /// <typeparam name="TM">measurement type</typeparam>
+        public async Task WriteMeasurementsAsync<TM>(string bucket, string org, WritePrecision precision,
+                        params TM[] measurements)
+        {
+            Arguments.CheckNonEmptyString(bucket, nameof(bucket));
+            Arguments.CheckNonEmptyString(org, nameof(org));
+            Arguments.CheckNotNull(precision, nameof(precision));
+
+            await WriteMeasurementsAsync(bucket, org, precision, measurements.ToList());
+        }
+
+        private async Task WriteData(string org, string bucket, BatchWriteData data)
+        {
+            var lineProtocol = data.ToLineProtocol();
+            var precision = data.Options.Precision;
+
+            await _service.PostWriteAsync(org, bucket, Encoding.UTF8.GetBytes(lineProtocol), null , 
+                            "identity", "text/plain; charset=utf-8", null, "application/json", null, precision);
+        }
```

Comment:
The write data should accept a list of BatchWriteData and process it in one request.

```suggestion
       private async Task WriteData(string org, string bucket, WritePrecision precision, IEnumerable<BatchWriteData> list)
        {
            var sb = new StringBuilder("");
            
            foreach (var item in list)
            {
                var lineProtocol = item.ToLineProtocol();
                
                if (string.IsNullOrEmpty(lineProtocol)) continue;

                sb.Append(lineProtocol);
                sb.Append("\n");
            }

            if (sb.Length == 0)
            {
                LOG nothing to write
                return;
            }
            
            // remove last \n
            sb.Remove(sb.Length - 1, 1);

            await _service.PostWriteAsync(org, bucket, Encoding.UTF8.GetBytes(sb.ToString()), null , 
                            "identity", "text/plain; charset=utf-8", null, "application/json", null, precision);
        }
```